### ISSUE TITLE
<strong> Tag Doesn't Shows Properly in <pre> Block

### ISSUE THREAD

**Issue by DawnMagnet** [facebook/docusaurus#6156]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

I Write a Test Case below
<pre><strong>输入：</strong>a = "abcd", b = "cdabcdab"
<strong>输出：</strong>3
<strong>解释：</strong>a 重复叠加三遍后为 "ab<strong>cdabcdab</strong>cd", 此时 b 是其子串。
</pre>
```html
<pre>
  <strong>输入：</strong>a = "abcd", b = "cdabcdab"
  <strong>输出：</strong>3
  <strong>解释：</strong>a 重复叠加三遍后为 "ab<strong>cdabcdab</strong>cd", 此时 b 是其子串。
</pre>
```
As we can see blow, Github shows correctly
but if this block appears in a docusaurus MarkDown file when build a website
it appears like blow
![image](https://user-images.githubusercontent.com/61567130/147038668-fa42990b-d9bd-49e1-b9b4-c97d8e71662f.png)
<pre tabindex="0" class="prism-code language-undefined codeBlock_rtdJ thin-scrollbar" style="color: rgb(57, 58, 52); background-color: rgb(246, 248, 250);"><code class="codeBlockLines_1zSZ"><span class="token-line" style="color: rgb(57, 58, 52);"><span class="token plain">[object Object]a = "abcd", b = "cdabcdab"[object Object]3[object Object]a 重复叠加三遍后为 "ab[object Object]cd", 此时 b 是其子串。</span><br></span></code></pre>
which turns `<strong>` tag into [object Object]

### Steps to reproduce

write a markdown file includes  `<strong>` tag in a `<pre>` tag
```html
<pre>
  <strong>输入：</strong>a = "abcd", b = "cdabcdab"
  <strong>输出：</strong>3
  <strong>解释：</strong>a 重复叠加三遍后为 "ab<strong>cdabcdab</strong>cd", 此时 b 是其子串。
</pre>
```

### Expected behavior

Correct Content

### Actual behavior

Incorrent

### Your environment

- Public source code: https://github.com/DawnMagnet/algorithm-station
- Public site URL:https://dawnmagnet.github.io/algorithm-station
- Docusaurus version used: latest
- Environment name and version (e.g. Chrome 89, Node.js 16.4): nodejs 14
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): travis-ci

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Looks like a Docusaurus error, can't reproduce in [MDX playground](https://mdx-git-renovate-babel-monorepo-mdx.vercel.app/playground)

**Comment by DawnMagnet** [comment]
> Looks like a Docusaurus error, can't reproduce in [MDX playground](https://mdx-git-renovate-babel-monorepo-mdx.vercel.app/playground). It's probably because we map `pre` elements to Code blocks. However, @DawnMagnet do take note that Docusaurus doesn't guarantee compatibility with GFM because we are, after all, using MDX. The line breaks aren't preserved even when using `pre` and you would probably have to use `{'\n'}` to preserve that

Thanks, I think MDX should have compatibility with native html, just use GFW to reproduce, as for the line breaks, I will use `<br/>` instead

**Comment by Josh-Cena** [comment]
> Thanks, I think MDX should have compatibility with native html

That would be impossible, because MDX is more biased to the JSX side than to the MD side. You should see it as JSX with MD interspersed rather than Markdown that happens to be rendered as JSX.

> GFW

I hope you mean GFM 😝

**Comment by DawnMagnet** [comment]
> That would be impossible, because MDX is more biased to the JSX side than to the MD side. You should see it as JSX with MD interspersed rather than Markdown that happens to be rendered as JSX.
Perhaps I should close this issue and figure out a new way to bold sth. in the code blocks?

**Comment by Josh-Cena** [comment]
Fixed by #6177. Please [test with the latest canary release](https://docusaurus.io/community/canary).
